comps/animation/wav2lip/dependency/Dockerfile

```diff
@@ -0,0 +1,76 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+# Use a base image
+FROM python:3.11-slim
+
+# Set environment variables
+ENV LANG=en_US.UTF-8
+ENV PYTHONPATH=/home:/home/user
+ENV PYTHON=/usr/bin/python3.11
+
+# Install dependencies
+RUN apt-get update -y && apt-get install -y --no-install-recommends --fix-missing \
+    yasm \
+    libx264-dev
+
+# TO-DO: Add a user for security
+
+# Install GenAIComps
+RUN mkdir -p /home/user/comps
+COPY comps /home/user/comps
+COPY comps/animation/dependency/entrypoint.sh /usr/local/bin/entrypoint.sh
+
+# Install ppmpeg with x264 software codec
+RUN git clone https://github.com/FFmpeg/FFmpeg.git /home/user/comps/animation/FFmpeg
+WORKDIR /home/user/comps/animation/FFmpeg
+RUN ./configure --enable-gpl --enable-libx264 && \
+    make -j$(nproc-1) && \
+    make install && \
+    hash -r
+RUN chmod +x $(which ffmpeg)
+
+# Upgrade pip
+RUN python3 -m pip install --upgrade pip
+
+# Install Wav2Lip from pip
+RUN pip install --no-deps Wav2Lipy
+RUN rm /usr/local/lib/python3.10/dist-packages/Wav2Lip/__init__.py && touch /usr/local/lib/python3.10/dist-packages/Wav2Lip/__init__.py
+ENV PYTHONPATH="/usr/local/lib/python3.10/dist-packages/Wav2Lip:$PYTHONPATH"
+
+# Install GFPGAN from pip
+RUN pip install --no-deps gfpgan
+RUN touch /usr/local/lib/python3.10/dist-packages/gfpgan/__init__.py
+ENV PYTHONPATH="/usr/local/lib/python3.10/dist-packages/gfpgan:$PYTHONPATH"
+
+# Download pre-trained models
+# Install gdown
+RUN pip install gdown
+WORKDIR /usr/local/lib/python3.10/dist-packages
+# RUN wget https://www.adrianbulat.com/downloads/python-fan/s3fd-619a316812.pth -O Wav2Lip/face_detection/detection/sfd/s3fd.pth
+RUN gdown https://drive.google.com/uc?id=1wwts8U4Wx2lpLOI9uGcUB-TLmKRIFnP8 -O Wav2Lip/face_detection/detection/sfd/s3fd.pth
```

Comment:
Since we download model file into docker image, do we have free redistribution license for those model files?

Reply:
Fixed by writing [download_ckpt.sh](https://github.com/ctao456/GenAIComps/blob/ctao/opea/comps/animation/wav2lip/dependency/download_ckpts.sh) and run it in Dockerfile, in the place of re-distribution.

Reply:
Thanks. Do you run download_ckpt.sh during docker run or docker build?  if it runs during docker build, we will still redistribute it.